README.md

```diff
@@ -1,12 +1,36 @@
-# lambda-template
-Template for creating lambda repositories
-A template for quickly starting a new AWS lambda project.
+# cfn-cr-sc-bucket-policy
 
-## Naming
-Naming conventions:
-* for a vanilla Lambda: `lambda-<context>`
-* for a Cloudformation Transform macro: `cfn-transform-<context>`
-* for a Cloudformation Custom Resource: `cfn-cr-<context>`
+Cloudformation Custom Resource that creates a S3 BucketPolicy. This is intended
+only for use in ServiceCatalog, as it retrieves an ARN that Service Catalog sets,
+`aws:servicecatalog:provisioningPrincipalArn`, to set as a principal on the
+BucketPolicy.
+
+Inventory of source code and supporting files:
+
+- policy_maker - Code for the application's Lambda function.
+- events - Invocation events that you can use to invoke the function.
+- tests - Unit tests for the application code.
+- template.yaml - A template that defines the application's AWS resources.
+
+## Use in a Cloudformation Template
+Create a custom resource in your cloud formation template. Here's an example:
+```yaml
+  SCS3BucketPolicy:
+    Type: Custom::SCS3BucketPolicy
+    Properties:
+      ServiceToken: !ImportValue
+        'Fn::Sub': '${AWS::Region}-cfn-cr-sc-bucket-policy-FunctionArn'
+      BucketName: !Ref S3Bucket
+      ExtraPrincipals: !Ref S3UserARNs
+      RequireEncryption: True
+```
+
+The creation of the custom resource triggers the lambda. It creates an S3
+BucketPolicy.
+* `ServiceToken` refers to the ARN of the lambda function. You can follow the pattern given; see "Install Lambda into AWS" below for the stack that exports that value.
+* The only required property is `BucketName`, a String.
+* `ExtraPrincipals` is an optional CommaDelimitedList of valid IAM policy [principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html) strings.
+* `RequireEncryption` is a boolean value; it will default to false if left out, otherwise, set it to `true` or `false` but do not quote those.
```

Comment:
I need to update the "quoted" statement -- I made it handle them whether quoted or not.

The policies currently used in `sc-s3-encrypted-ra-v1.0.0.yaml` and `sc-s3-synapse-ra-v1.0.0.yaml` share the same first two statements, the only difference being in what principals are added (the first is `Sid: ReadAccess`, the second is `Sid: WriteAccess`). `sc-s3-encrypted-ra-v1.0.0.yaml` has two additional statements: `Sid: DenyIncorrectEncryptionHeader` and `Sid: DenyUnEncryptedObjectUploads`.  The policies as they currently exist can be handled by this custom resource in both cases through setting properties. You will see a PR later today for that.

Regarding your questions about BucketEncrypt and CORS rules: incorrect, the lambda is not setting anything in regard to that. Those are set in the bucket properties. Specifically, at this [line](https://github.com/Sage-Bionetworks/service-catalog-library/blob/master/s3/sc-s3-encrypted-ra-v1.0.0.yaml#L25) for the encrypted bucket and at this [line](https://github.com/Sage-Bionetworks/service-catalog-library/blob/master/s3/sc-s3-synapse-ra-v1.0.0.yaml#L39) and this [line](https://github.com/Sage-Bionetworks/service-catalog-library/blob/master/s3/sc-s3-synapse-ra-v1.0.0.yaml#L43) for the synapse bucket.

None of this conflicts since the custom resource is not setting any bucket properties, it is creating policies, and the policies exactly match what's already there.

Reply:
Revised "quoted" in README in commit f6f9d8b

Reply:
To see the updated product templates using the new custom resource, see this [PR](https://github.com/Sage-Bionetworks/service-catalog-library/pull/151/files) or take a look at the products in my S3 portfolio in the dev environment.